From: Tariq Toukan <tariqt@mellanox.com>
Date: Mon, 9 Oct 2017 16:59:50 +0300
Subject: net/mlx4_en: Use __force to fix a sparse warning in TX datapath
Patch-mainline: v4.15-rc1
Git-commit: 7ba5e7bd64a5715f624d39815f567842178afb72
References: bug#925105

In TX data-path, we intentionally do not byte-swap, as documented
in code and in the cited commit log.
This fixes sparse warning:
en_tx.c:720:23: warning: incorrect type in argument 1 (different base types)
en_tx.c:720:23:    expected unsigned int [unsigned] [usertype] <noident>
en_tx.c:720:23:    got restricted __be32 [usertype] doorbell_qpn

Fixes: 492f5add4be8 ("net/mlx4_en: Doorbell is byteswapped in Little Endian archs")
Signed-off-by: Tariq Toukan <tariqt@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Benjamin Poirier <bpoirier@suse.com>
---
 drivers/net/ethernet/mellanox/mlx4/en_tx.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/ethernet/mellanox/mlx4/en_tx.c
+++ b/drivers/net/ethernet/mellanox/mlx4/en_tx.c
@@ -947,7 +947,7 @@ netdev_tx_t mlx4_en_xmit(struct sk_buff
 #else
 		iowrite32be(
 #endif
-		          ring->doorbell_qpn,
+			  (__force u32)ring->doorbell_qpn,
 			  ring->bf.uar->map + MLX4_SEND_DOORBELL);
 	}
 
